mungegithub/mungers/sig-mention-handler_test.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/github"
+	github_test "k8s.io/test-infra/mungegithub/github/testing"
+	"net/http"
+	"reflect"
+	"runtime"
+	"testing"
+)
+
+const (
+	helpWanted    = "help-wanted"
+	open          = "open"
+	sigApps       = "sig/apps"
+	username      = "Ali"
+	needsSigLabel = "needs-sig"
+)
+
+func TestSigMentionHandler(t *testing.T) {
+	runtime.GOMAXPROCS(runtime.NumCPU())
+
+	tests := []struct {
+		name     string
+		issue    *githubapi.Issue
+		expected []githubapi.Label
+	}{
+		{
+			name: "ignore PRs",
+			issue: &githubapi.Issue{
+				State: githubapi.String(open),
+				Labels: []githubapi.Label{githubapi.Label{Name: githubapi.String(helpWanted)},
```

Comment:
Why we want to sigApps labels if they are already there? For example, an issue has a `sig/apps` label. with my understanding, the issue should be ignored to prevent adding the `needs-sig` label.